qtop.py

```diff
@@ -486,26 +516,56 @@ def calculate_remaining_matrices(node_state,
     and the remaining 190 machines have 8 cores, this doesn't print the non-existent
     56 cores from the next matrix on.
     """
-    gray_hash = '\x1b[1;30m#\x1b[1;m'
-    separator_between_ansi = '\x1b[0m|\x1b[1;m'
-
+    # need node_state, temp
     for matrix in range(extra_matrices_nr):
-        print_start = _print_end
-        if config['user_cut_matrix_width']:
-            _print_end += config['user_cut_matrix_width']
+        print_start = print_end
+        if USER_CUT_MATRIX_WIDTH:
+            print_end += USER_CUT_MATRIX_WIDTH
         else:
-            _print_end += term_columns - DEADWEIGHT
-        _print_end = min(_print_end, node_dict['total_wn']) if options.REMAP else min(_print_end, node_dict['highest_wn'])
+            print_end += term_columns - DEADWEIGHT
+        print_end = min(print_end, cluster_dict['total_wn']) if options.REMAP else min(print_end, cluster_dict['highest_wn'])
+
+        display_selected_occupancy_parts(print_start, print_end, wn_vert_labels, core_user_map, pattern_of_id, wn_occup)
+
+
+def display_selected_occupancy_parts(print_start, print_end, wn_vert_labels, core_user_map, pattern_of_id, wn_occup):
+    """
+    occupancy_parts needs to be redefined for each matrix, because of changed parameter values
+    """
+    occupancy_parts = {
+        'wn id lines': (print_wnid_lines, (print_start, print_end, cluster_dict['highest_wn'], wn_vert_labels),
+                        {'inner_attrs': None}),
+        'core user map': (print_core_lines, (core_user_map, print_start, print_end, pattern_of_id), {'attrs': None}),
+        # 'node state': (print_single_attr_line, (print_start, print_end), {'attr_line': wn_occup['node state']}),
+        # 'temperature': (print_single_attr_line, (print_start, print_end), {'attr_line': wn_occup['temperature']}),
+    }
+    for yaml_key, part_name in get_yaml_key_part('wn_occupancy'):
+        occupancy_parts.update(
+            {part_name: (print_single_attr_line, (print_start, print_end), {'attr_line': wn_occup[part_name]})}
+        )
```

Comment:
for readability: collapse three lines into two, by assigning fi. `newdictvar` 'h kapoio kalytero onoma)